### DISCUSSION TITLE
How does one empty the cache?

### DISCUSSION THREAD

**Discussion by mrflip** [urql-graphql/urql#1236] [Q&A]
Our app (expo for react native and next-js for react, single codebase) is intended to be multi-org — one user can belong to multiple organizations, each with completely silo'ed data. When that user switches orgs, we'd like to flush and fill the cache. 

Sorry if this is obvious, but is that a first-class operation in the graphcache API?

**Comment by arcanis** [comment]
I think you're supposed to create a new client, which will come with a fresh cache.

**Reply by kitten** [reply]
Yes, that's correct, we recommend to create the client again and to swap it out. Each framework binding, including the React one, knows how to deal with a changed client and will update automatically.

The reason why we recommend this is that there are a lot of different states exchanges can collect, for instance some keep track of active operations. The safest way to ensure that everything is properly cleaned up is to recreate the client or use different ones to separate this state.
